jdaviz/core/template_mixin.py

```diff
@@ -335,7 +339,27 @@ def __init__(self, **kwargs):
         # in repeated toggling of is_active.  To use, decorate any method that observes traitlet
         # changes (including is_active) with @skip_if_no_updates_since_last_active()
         self._methods_skip_since_last_active = []
-        super().__init__(**kwargs)
+
+        # get default viewer names from the helper, according to the requirements of the plugin
+        for registry_name, tray_item in tray_registry.members.items():
+            if tray_item['cls'] == self.__class__:
+                # If viewer reference names need to be passed to the tray item
+                # constructor, pass the names into the constructor in the format
+                # that the tray items expect.
+                tray_registry_options = tray_item.get('viewer_reference_name_kwargs', {})
+                for opt_attr, [opt_kwarg, get_name_kwargs] in tray_registry_options.items():
+                    opt_value = getattr(
+                        self, opt_attr, app._get_first_viewer_reference_name(**get_name_kwargs)
+                    )
+
+                    if opt_value is None:
+                        continue
```

Comment:
this for-loop is ugly.  I have a local stash that attempts to replace this by having the registry itself set `cls._viewer_reference_name_kwargs` which shows promise but still needs some work.